randovania/discord_bot/preset_lookup.py

```diff
@@ -260,16 +263,16 @@ async def _try_get[T](
         return None
 
 
-async def _get_presets_from_message(message: discord.Message):
+async def _get_presets_from_message(message: discord.Message) -> AsyncGenerator[Preset[BaseConfiguration]]:
     for attachment in message.attachments:
         filename: str = attachment.filename
         if filename.endswith(VersionedPreset.file_extension()):
-            result = await _try_get(message, attachment, VersionedPreset)
+            result = await _try_get(message, attachment, lambda d: VersionedPreset(d).get_preset())
```

Comment:
mypy wants type annotation here. Maybe similar bug to what was mentioned in #8534 ?

Reply:
In this case it's tricky because we don't have `Preset` defaulting to `Preset[BaseConfiguration]` (I think?) so it really needs a hint.